ember-headlessui/addon/components/disclosure/button.hbs

```diff
@@ -0,0 +1,12 @@
+<button
+  type='button'
+  aria-haspopup={{true}}
+  aria-controls={{if @isOpen this.itemsGuid}}
```

Comment:
can we generate a single GUID in the parent component, just to reduce the number of repeat guid creations?

passing a guid (and then concatting suffixes) allows us to then also get rid of data-test-selectors (we don't want to ship these to consumers either, but that's a current problem all over the addon that we need to fix -- out of scope for this PR)